src/librbd/librbd.cc

```diff
@@ -2340,6 +2340,11 @@ extern "C" int rbd_trash_list(rados_ioctx_t p, rbd_trash_image_info_t *entries,
 
   vector<librbd::trash_image_info_t> cpp_entries;
   int r = librbd::trash_list(io_ctx, cpp_entries);
+  if (r == -ENOENT) {
```

Comment:
OK, i will address it, and all other list apis, e.g., `RBD::list`, `RBD::group_list`, etc.